Extraction/Extractor/PhpDocOperationExtractor.php

```diff
@@ -41,49 +62,36 @@ public function extract($method, $operation, ExtractionContextInterface $extract
         $factory = DocBlockFactory::createInstance();
         $docBlock = $factory->create($method->getDocComment());
 
-        if (!$operation->summary) {
+        if(!$operation->summary) {
             $operation->summary = $docBlock->getSummary();
         }
 
-        if ($operation->description) {
+        if($operation->description) {
             $operation->description = $docBlock->getDescription();
         }
 
-        $returnTags = $docBlock->getTagsByName('return');
-        if (empty($returnTags)) {
+        foreach ($docBlock->getTagsByName('return') as $returnTag) {
+            /* @var $returnTag \phpDocumentor\Reflection\DocBlock\Tags\Return_ */
             $response = new Response();
             $response->schema = $responseSchema = new Schema();
-            $response->description = "Generic 200 response";
+            $response->description = $returnTag->getDescription();
             $operation->responses[200] = $response;
+
             $subContext = $extractionContext->createSubContext();
             $subContext->setParameter('direction', 'out');
-            $extractionContext->getSwagger()->extract("test", $responseSchema, $subContext);
-
-        } else {
-            /** @var DocBlock\Tags\Return_ $returnTag */
-            foreach ($returnTags as $returnTag) {
-                /** @var Object_ $type */
-                $type = $returnTag->getType();
-                $response = new Response();
-                $response->schema = $responseSchema = new Schema();
-                $response->description = $returnTag->getDescription();
-                $operation->responses[200] = $response;
-
-                $subContext = $extractionContext->createSubContext();
-                $subContext->setParameter('direction', 'out');
-
-                $extractionContext->getSwagger()->extract((string)$type, $responseSchema, $subContext);
-            }
+
+            $extractionContext->getSwagger()->extract($returnTag->getType(), $responseSchema, $subContext);
         }
 
-        if ($docBlock->getTagsByName('deprecated')) {
-            $operation->deprecated = true;
+        if($docBlock->getTagsByName('deprecated')) {
+           $operation->deprecated = true;
         }
 
-        /* @var $throwTag \phpDocumentor\Reflection\DocBlock\Tags\Throws */
         foreach ($docBlock->getTagsByName('throws') as $throwTag) {
-            /** @var Object_ $type */
+            /* @var $throwTag \phpDocumentor\Reflection\DocBlock\Tags\Throws */
+
```

Comment:
I would remove the empty line to make clear this type hint belongs to the line below.